bluepysnap/edges.py

```diff
@@ -24,94 +24,34 @@
 from cached_property import cached_property
 from more_itertools import first
 
+from bluepysnap.network import NetworkObject
 from bluepysnap.exceptions import BluepySnapError
 from bluepysnap.circuit_ids import CircuitEdgeId, CircuitEdgeIds, CircuitNodeId, CircuitNodeIds
 from bluepysnap.sonata_constants import DYNAMICS_PREFIX, Edge, ConstContainer
 from bluepysnap import utils
+from bluepysnap.utils import Deprecate
+from bluepysnap._doctools import AbstractDocSubstitutionMeta
 
 
-class Edges:
+class Edges(NetworkObject, metaclass=AbstractDocSubstitutionMeta,
+            source_word="NetworkObject", target_word="Edge"):
     """The top level Edges accessor."""
 
-    def __init__(self, circuit):
+    def __init__(self, circuit):  # pylint: disable=useless-super-delegation
         """Initialize the top level Edges accessor."""
-        self._circuit = circuit
-        self._config = self._circuit.config['networks']['edges']
-        self._populations = self._get_populations()
+        super().__init__(circuit)
 
     def _get_populations(self):
```

Comment:
I really hoped that `_get_populations`,`ids`,`get` would be factorized

Reply:
I must say I did not find a good way of doing it...

The problem are the specific classes used in these functions : EdgeStorage, CircuitEdgeIds, CircuitEdgeId.

This is indeed easy for the get.

Reply:
I made something but this is not quite perfect.